Server/API/RemoteControlController.cs

```diff
@@ -114,17 +118,17 @@ bool remoteControlStarted()
                 {
                     return StatusCode(408, "The remote control process failed to start in time on the remote device.");
                 }
-                else
-                {
-                    var rcSession = CasterHub.SessionInfoList.Values.LastOrDefault(x => x.DeviceID == targetDevice.Value.ID && !existingSessions.Contains(x.CasterSocketID));
-                    var otp = RemoteControlFilterAttribute.GetOtp(targetDevice.Value.ID);
-                    return Ok($"{HttpContext.Request.Scheme}://{Request.Host}/RemoteControl?casterID={rcSession.CasterSocketID}&serviceID={targetDevice.Key}&fromApi=true&otp={Uri.EscapeDataString(otp)}");
-                }
             }
             else
             {
-                return BadRequest("The target device couldn't be found.");
+                var rcSession = CasterHub.SessionInfoList.Values.LastOrDefault(x => x.DeviceID == deviceID);
+                if (rcSession != null)
+                {
+                    var otp = RemoteControlFilterAttribute.GetOtp(deviceID);
```

Comment:
We'd want to verify that user has authorization to access device with `DataService.DoesUserHaveAccessToDevice`.